---
title: "figure-z"
author: "Mac Campbell"
date: "3/22/2022"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning=FALSE, message=FALSE)
```

```{r}
library(tidyverse)
library(ape)
library(ggnetworx)
library(viridis)
library(phangorn)
library(tanggle)
```

### Get network
```{r}
network<-read.nexus.networx(file = "figure-6/84/splits-network.nex")
```


### Get Meta

```{r}
data<-read_tsv("data/pcaSc75.edited.clst") %>% select(FID, Location) %>%
  mutate(Color = ifelse(Location %in% c("PITMC","PITRC"), "#ACCC66", #Pit
                           ifelse(Location %in% c("CLS","McC M","NAP","PUT BC","PUT RR","SUS","WAD"), "grey50",   #Prickly
                           ifelse(Location %in% c("SAG"),"grey50", # Paiute
                           ifelse(Location %in% c("COY","UVAS","PEN","GUAD"),"#E8BEFF", # C o o
                           ifelse(Location %in% c("RUSEF","RUS"), "#EEF1A0", # C o p
                           ifelse(Location %in% c("MER","MOK"), "#DE9E66", # C g g
                           ifelse(Location %in% c("SAC","McC HB","HSS"), "#439AB0", "grey50")))))))) %>%
  mutate(Taxon = ifelse(Location %in% c("PITMC","PITRC"), "C. pitensis", #Pit
                           ifelse(Location %in% c("CLS","McC M","NAP","PUT BC","PUT RR","SUS","WAD"), "Other",   #Prickly
                           ifelse(Location %in% c("SAG"),"Other", # Paiute
                           ifelse(Location %in% c("COY","UVAS","PEN","GUAD"),"C. o. ohlone", # C o o
                           ifelse(Location %in% c("RUSEF","RUS"), "C. o. pomo", # C o p
                           ifelse(Location %in% c("MER","MOK"), "C. g. gulosus", # C g g
                           ifelse(Location %in% c("SAC","McC HB","HSS"), "C. g. wintu", "Other")))))))) %>%
  mutate(Names = ifelse(Location %in% c("PITMC","PITRC"), "C. pitensis", #Pit
                           ifelse(Location %in% c("CLS","McC M","NAP","PUT BC","PUT RR","SUS","WAD"), "C. asper",   #Prickly
                           ifelse(Location %in% c("SAG"),"C. beldingii", # Paiute
                           ifelse(Location %in% c("COY","UVAS","PEN","GUAD"),"C. o. ohlone", # C o o
                           ifelse(Location %in% c("RUSEF","RUS"), "C. o. pomo", # C o p
                           ifelse(Location %in% c("MER","MOK"), "C. g. gulosus", # C g g
                           ifelse(Location %in% c("SAC","McC HB","HSS"), "C. g. wintu", "Other"))))))))
```
```{r}
tips<-as_tibble(network$tip.label) %>% rename(FID=value) %>% left_join(data)
```


```{r}
pdf(file="figure-z/network.pdf", width=11, height=8)

plot(network, tip.color = tips$Color, edge.width = 0.5, cex=0.5)
dev.off()
```

```{r}

library(phangorn)
dna<-read.phyDat("figure-6/84/miss10-noogs.phy")
dat<-as.phyDat(dna)
set.seed(1)
bs <- bootstrap.phyDat(dat, FUN = function(x)nj(dist.hamming(x)), 
    bs=100)
tree <- nj(dist.hamming(dat))
#tree<-root(tree, c("Put_RR_A2\t","McC_M_C7\t","SagS4_2\t","SagS2_2\t"))
plot(tree)

```

```{r}
par("mar" = rep(1, 4))
tree <- plotBS(tree, bs, "phylogram")

#By default prob=0.3
cnet <- consensusNet(bs, .25)
plot(cnet, "2D", show.edge.label=TRUE)
```

```{r}
cnet$tip.label<-gsub("\t","",cnet$tip.label)

#plot(cnet, "2D", show.edge.label=TRUE)

edge.col <- createLabel(cnet, tree, "black", nomatch="grey")

plot(cnet, show.edge.label = T, "2D", edge.color = edge.col,
                  col.edge.label = "blue", cex=.75)
```

```{r}
plot(cnet, show.edge.label = F, "2D", edge.color = edge.col,
                  col.edge.label = "blue", cex=.75)
```

```{r}
ddf<-cnet$tip.labe %>% as_tibble() %>% left_join(data, by=c("value"="FID"))
```

```{r}
pdf("figure-z/cnet.pdf")
plot(cnet, show.edge.label = F, "2D", edge.color = edge.col, tip.color = ddf$Color,
                  col.edge.label = "blue", cex=.75,
     tip.label=ddf$Location)
dev.off()
```

Bootstrapping with RAXML

```{sh, eval=FALSE}
iqtree -s miss10.phy -st DNA -m MFP+ASC -bb 1000 -alrt 1000 -redo

raxmlHPC-PTHREADS -T 2 -m ASC_GTRGAMMA --asc-corr=lewis -s miss10.phy.varsites.phy -p 123 -f a -x 100 -# 100 -n snps

```


```{r}
## RAxML best-known tree with bipartition support (from previous analysis)
raxml.tree <- read.tree("figure-z/RAxML_bipartitions.snps")
## RAxML bootstrap trees (from previous analysis)
raxml.bootstrap <- read.tree("figure-z/RAxML_bootstrap.snps")

plot(raxml.tree)
nodelabels(raxml.tree$node.label, adj = c(1, 0), frame = "none")
```


Let's try the whatever, treemix

## Treemix

A text file with no header line, and one line per sample with the following three fields:

1 Family ID 2 Within-family ID 3 Cluster name

`(base) Macs-MBP-2:figure-z mac$ cp ../figure-6/84/recode.prune.reheadered.vcf ./84-recode.prune.reheadered.vcf`
`cat ../figure-6/84/samples.tsv | awk '{print $1, $1}' > 84-part1.txt`
`(base) Macs-MBP-2:figure-z mac$ cat ../figure-6/84/samples.tsv | cut -f 1 -d '_' > 84-part2.txt`
`(base) Macs-MBP-2:figure-z mac$ paste 84-part1.txt 84-part2.txt  > 84.clst`

`(base) Macs-MBP-2:figure-z mac$ 
cat 84-recode.prune.reheadered.vcf | perl -pe 's/^R0/chr/g'| perl -pe 's/ID=R0/ID=chr/g' > redoneChroms.vcf
    
Some tidying up now working in /temp/
then a problem with chroms... https://www.biostars.org/p/109690/

bcftools view -H redoneChroms.vcf | cut -f 1 | uniq | awk '{print $0"\t"$0}' > redoneChroms.chrom-map.txt

vcftools --vcf  redoneChroms.vcf --plink --mac 2 --remove-indels --max-alleles 2  --out redoneChroms --chrom-map redoneChroms.chrom-map.txt
plink --file redoneChroms --make-bed --out redoneChroms --allow-no-sex --allow-extra-chr # drop 0
plink --bfile redoneChroms --freq --missing  --within 84.clst --out redoneChroms --allow-no-sex --allow-extra-chr 
 
seems fine now.

gzip redoneChroms.frq.strat

plink2treemix.py redoneChroms.frq.strat.gz redoneChroms.treemix.frq.gz

Our freq file now has info in it.
treemix -i redoneChroms.treemix.frq.gz -m 0 -o redone-0 -root beldingii -bootstrap -k 10 -noss > treemix_0_log 
treemix -i redoneChroms.treemix.frq.gz -m 1 -o redone-1 -root beldingii -bootstrap -k 10 -noss > treemix_1_log 
treemix -i redoneChroms.treemix.frq.gz -m 2 -o redone-2 -root beldingii -bootstrap -k 10 -noss > treemix_2_log 

treemix -i redoneChroms.treemix.frq.gz -m 0 -o redone-noss-0 -root beldingii -bootstrap -k 10 -noss > treemix_0_log 
treemix -i redoneChroms.treemix.frq.gz -m 1 -o redone-noss-1 -root beldingii -bootstrap -k 10 -noss > treemix_1_log 
treemix -i redoneChroms.treemix.frq.gz -m 2 -o redone-noss-2 -root beldingii -bootstrap -k 10 -noss > treemix_2_log 
treemix -i redoneChroms.treemix.frq.gz -m 3 -o redone-noss-3 -root beldingii -bootstrap -k 10 -noss > treemix_3_log 


```{sh,eval=FALSE}
for i in {0..2}
do
 treemix -i 84-recode.prune.reheadered.treemix.frq.gz -m $i -o 84-$i -root beldingii -bootstrap -k 500 > treemix_${i}_log &
done
```
treemix -i 84-recode.prune.reheadered.treemix.frq.gz -m 1 -o 84-1 -root beldingii -bootstrap -k 1 -noss > treemix_1_log 


  did it work?
  
No, because vcftools throws out all our chroms.

```{r}
library(RColorBrewer)
library(R.utils)
source("~/treemix/treemix-1.13/src/plotting_funcs.R")

```


```{r}
plot_tree(cex=0.8, "figure-z/temp/redone-0")
```

```{r}
getwd()

prefix="figure-z/temp/redone-"

par(mfrow=c(1,3))
for(edge in 0:2){
  plot_tree(cex=0.8,paste0(prefix,edge))
  title(paste(edge,"edges"))
}
```

```{r}

prefix="figure-z/temp/redone-noss-"

par(mfrow=c(1,3))
for(edge in 0:2){
  plot_tree(cex=0.8,paste0(prefix,edge))
  title(paste(edge,"edges"))
}
```


```{r}
pdf("figure-z/treemix.pdf")

prefix="figure-z/temp/redone-noss-"

par(mfrow=c(2,2))
for(edge in 0:2){
  plot_tree(cex=0.7,paste0(prefix,edge))
  title(paste(edge,"edges"))
}
```
