---
title: "Figure-6"
author: "Mac Campbell"
date: "3/1/2022"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning=FALSE, message=FALSE)
```

```{r}
library(tidyverse)
library(ggtree)
```

## Phylogeny

Generating a concatenated ML phylogeny matching sampling PCA of all species, i.e.  bamlist_75.

Setting up a sensible path `cat bamlist_75 | awk {'print "bams/"$1'} > bamlist_75.bamlist`    

__1__ Calling SNPs     

```{sh, eval=FALSE}
srun -p high -t 48:00:00 --mem=16G --nodes=1 --ntasks=1 angsd -P 2 \
-bam bamlists/bamlist_75.bamlist \
-out figure-6/plink  \
-minInd 105 -minMaf 0.05  -minMapQ 20 -minQ 20 -GL 1 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doGeno 4 -doPost 1 -postCutoff 0.95 -doPlink 2 > figure-6/std.out 2> figure-6/std.err &
```

__2__ Convert to vcf and prune (local)     

```{sh, eval=FALSE}
plink --tped plink.tped --tfam plink.tfam  --out binary --recode --allow-extra-chr --noweb
plink --ped binary.ped --map binary.map --recode vcf --allow-extra-chr -out recode
bcftools +prune -l 0.9 -w 10000 recode.vcf  -Ov -o recode.prune.vcf
```

__3__ Get some sensible names

```{sh, eval=FALSE}
cat ../data/pcaSc75.edited.clst | cut -f 2 | grep -v FID  > samples.tsv
bcftools reheader --samples samples.tsv -o recode.prune.reheadered.vcf recode.prune.vcf
```

__4__ Prepare for ML     

```{sh, eval=FALSE}
source activate py2;  python2.7 ~/github/mccloud-rrt/vcf2phylip.py -i recode.prune.reheadered.vcf; conda deactivate;

conda activate py3; ~/github/mccloud-rrt/103-remove-invariant.py -p recode.prune.reheadered.min4.phy -o recode.prune.reheadered.min4.asc.phy; conda deactivate;
```

Remove missing data.
```{sh,eval=FALSE}
../201.1-removeMissing.pl recode.prune.reheadered.min4.asc.phy > miss10.phy
seqConverter.pl -dmiss10.phy -on
```

__5__ ML     

```{sh,eval=FALSE}
iqtree -s miss10.phy -st DNA -m MFP+ASC -bb 1000 -alrt 1000 -redo
iqtree -s miss10.phy.varsites.phy -st DNA -m MFP+ASC -bb 1000 -alrt 1000 -redo
```

## Plot

```{r}
tree<-read.tree("figure-6/miss10.phy.varsites.phy.contree")
```


```{r}
data<-read_tsv("data/pcaSc75.edited.clst") %>% select(FID, Location) %>%
  mutate(Color = ifelse(Location %in% c("PITMC","PITRC"), "#ACCC66", #Pit
                           ifelse(Location %in% c("CLS","McC M","NAP","PUT BC","PUT RR","SUS","WAD"), "grey50",   #Prickly
                           ifelse(Location %in% c("SAG"),"grey50", # Paiute
                           ifelse(Location %in% c("COY","UVAS","PEN","GUAD"),"#E8BEFF", # C o o
                           ifelse(Location %in% c("RUSEF","RUS"), "#EEF1A0", # C o p
                           ifelse(Location %in% c("MER","MOK"), "#DE9E66", # C g g
                           ifelse(Location %in% c("SAC","McC HB","HSS"), "#439AB0", "grey50")))))))) %>%
  mutate(Taxon = ifelse(Location %in% c("PITMC","PITRC"), "C. pitensis", #Pit
                           ifelse(Location %in% c("CLS","McC M","NAP","PUT BC","PUT RR","SUS","WAD"), "Other",   #Prickly
                           ifelse(Location %in% c("SAG"),"Other", # Paiute
                           ifelse(Location %in% c("COY","UVAS","PEN","GUAD"),"C. o. ohlone", # C o o
                           ifelse(Location %in% c("RUSEF","RUS"), "C. o. pomo", # C o p
                           ifelse(Location %in% c("MER","MOK"), "C. g. gulosus", # C g g
                           ifelse(Location %in% c("SAC","McC HB","HSS"), "C. g. wintu", "Other")))))))) %>%
  mutate(Names = ifelse(Location %in% c("PITMC","PITRC"), "C. pitensis", #Pit
                           ifelse(Location %in% c("CLS","McC M","NAP","PUT BC","PUT RR","SUS","WAD"), "C. asper",   #Prickly
                           ifelse(Location %in% c("SAG"),"C. beldingii", # Paiute
                           ifelse(Location %in% c("COY","UVAS","PEN","GUAD"),"C. o. ohlone", # C o o
                           ifelse(Location %in% c("RUSEF","RUS"), "C. o. pomo", # C o p
                           ifelse(Location %in% c("MER","MOK"), "C. g. gulosus", # C g g
                           ifelse(Location %in% c("SAC","McC HB","HSS"), "C. g. wintu", "Other"))))))))

```
```{r}
t<-ggtree(tree, layout = "daylight") %<+% data 
t + geom_tippoint(aes(fill=Location), pch=21)
```

```{r}
t$data$Bootstrap<-as.numeric(t$data$label)

d <- t$data
d <- d[!d$isTip,]
d$label <- as.numeric(d$label)
d <- d[d$label > 90,]

e <- t$data
e <- e[!e$isTip,]
e$label <- as.numeric(e$label)
e <- e[e$label < 90,]
e <- e[e$label > 75,]

```

```{r}
t +
  geom_point(data=e,  color="gray50", cex=3, alpha=0.75, pch=18) +
  geom_point(data=d,  color="black", cex=3, alpha=0.75, pch=18) +
  geom_tippoint(aes(fill=Taxon), pch=21, cex=4) +
  scale_shape_manual(values=c(21), guide=FALSE) +
  scale_fill_manual(values=c("#DE9E66","#439AB0","#E8BEFF","#EEF1A0","#ACCC66", "grey50"))

#  scale_fill_manual(values=t$data$Color) +
#  guides(fill = guide_legend(override.aes=list(shape=21))) 
#  ggtitle("C") +
#  theme(plot.title = element_text(size=16, face="bold"))

```

```{r}
ggtree(tree) %<+% data +
  geom_tiplab(aes(color=Taxon), cex=3, align=TRUE) +
  scale_shape_manual(values=c(21), guide=FALSE) +
  scale_color_manual(values=c("#DE9E66","#439AB0","#E8BEFF","#EEF1A0","#ACCC66", "grey50")) +
  xlim(0,0.2)

ggsave("figure-6/basic-tree.pdf", height=10, width=6)

```

```{r}
library(ape)
library(phangorn)
```

```{r}
dat<-read.nexus.data(file="figure-6/miss10.nex")
write.nexus.data(dat, file="figure-6/miss10-forsplits.nex")
```


### Reducing outgroup taxa:

We can prune down the Pit and Prickly sculpins and get more data!! Keeping 2 of each for species tree.

Getting counts:
```{sh, eval=FALSE}
ls | grep sort.flt.bam | grep -v bai | while read line; do samtools flagstat $line | grep mapped | head -n 1 >> counts.txt; done;
ls | grep sort.flt.bam | grep -v bai >> counts.files.txt
```

```{r}
files<-read_tsv("figure-6/counts.files.txt", col_names="File")
counts<-read_tsv("figure-6/counts.txt", col_names="Counts")
counts$Counts<-gsub(" + 0 mapped (100.00% : N/A)", "", counts$Counts, fixed = TRUE)

counted<-bind_cols(files,counts)
counted$`Sequence File Name`<-gsub("_RA.sort.flt.bam","",counted$File)
counted$Counts<-as.numeric(counted$Counts)
```
```{r}
comb<-left_join(data, counted, by=c("FID"="Sequence File Name"))
```


Get top outgroups:

```{r}
outs<-comb %>% filter(Names %in% c("C. asper","C. beldingii")) %>% group_by(Names) %>% arrange(-Counts) %>% top_n(2)
outs
```

Drop other outgroups:

```{r}
dnew<-comb %>% filter(Taxon !="Other") %>% bind_rows(outs) %>% mutate(Path=paste0("bams/",File))

write_tsv(dnew %>% select(Path), "bamlists/bamlist-84.bamlist")
```


```{sh, eval=FALSE}
srun -p high -t 48:00:00 --mem=16G --nodes=1 --ntasks=1 angsd -P 2 \
-bam bamlists/bamlist-84.bamlist \
-out figure-6/plink84  \
-minInd 80 -minMaf 0.05  -minMapQ 20 -minQ 20 -GL 1 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doGeno 4 -doPost 1 -postCutoff 0.95 -doPlink 2 > figure-6/std84.out 2> figure-6/std84.err &
```
